packages/driver/src/primitives/buffer.ts

```diff
@@ -19,28 +19,49 @@
 import type char from "./chars";
 import * as chars from "./chars";
 import { LegacyHeaderCodes } from "../ifaces";
-import { Buffer } from "../adapter.node";
-
-/* WriteBuffer over-allocation */
-const BUFFER_INC_SIZE = 4096;
-
-const EMPTY_BUFFER = new Uint8Array(0);
 
 export const utf8Encoder = new TextEncoder();
 export const utf8Decoder = new TextDecoder("utf8");
 
-const decodeB64 = (b64: string): Uint8Array => {
-  return Buffer.from(b64, "base64");
-};
-const encodeB64 = (data: Uint8Array): string => {
-  const buf = Buffer.isBuffer(data)
-    ? data
-    : Buffer.from(data.buffer, data.byteOffset, data.byteLength);
-  return buf.toString("base64");
-};
+let decodeB64: (_: string) => Uint8Array;
+let encodeB64: (_: Uint8Array) => string;
+
+try {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  const Buffer = require("node:buffer").Buffer;
```

Comment:
The tests are failing because we cannot use `require` in Deno, and Deno will unconditionally try to run this code. We typically can get around this by putting the non-Deno stuff in the `adapter.node.ts` module, and making a Deno-compatible import available in `adapter.deno.ts`, and at `buildDeno`-time, we swap these out in the source. This is what we did before by exporting `Buffer` from the adapter.

I _think_ we should be able to do that if we move these base64-related functions into the `adapter.*` modules and import them instead of importing `Buffer`. And having the `adapter.node.ts` module do this check instead.

Reply:
I don't think this will work in the browser because then we go back to importing things from adapter.node which will try importing node specific libs that are not valid in the browser.

Reply:
this will probably not work because we will import from adapter.node and that file cant be read in browser

Reply:
yeah, that's true. I didn't really consider how this module before was totally runtime-agnostic, where the adapter stuff was always imported into server-only contexts.

> https://github.com/edgedb/edgedb-js/issues/1003 the "Buffer not defined" err was thrown because on line 35 typeof was missing: if (Buffer === "function") {. So native node Buffer was never used until new PR that fixes that but broke browser functioning.

Let's just adopt this where we fix the check, and keep most of the old code. I think a more thorough fix will come when we make platform-specific packages so the modules don't need to be so dynamic, but we are still a ways off from working on that.

Reply:
@scotttrinh @jaclarke I think this should be it. Btw I didnt add any new tests, can do that too.